internal/tofu/context_apply_test.go

```diff
@@ -7225,8 +7224,8 @@ func TestContext2Apply_targetedDestroy(t *testing.T) {
 
 	// the module instance should remain
 	mod = state.Module(addrs.RootModuleInstance.Child("child", addrs.NoKey))
-	if len(mod.Resources) != 1 {
-		t.Fatalf("expected 1 resources, got: %#v", mod.Resources)
+	if mod != nil {
```

Comment:
This test was wrong

Reply:
Could you explain how this test was wrong?

Reply:
Basically, the state was built in such a way were dependencies were wrong. The actual configuration file marks the module as dependent on the standalone resource. However, this was not reflected in the state file.

By actually simulating an apply instead of building the state file (which some tests do), I was able to get the actual expected behaviour here

Changed a comment here:
82b9d01d245dae95a021f87cd2041cb843f326ef